.github/actions/install-deps/action.yml

```diff
@@ -0,0 +1,31 @@
+name: Install backend dependencies
+description: Installs dependencies for python backend
+
+runs:
+  using: composite
+  steps:
+    - name: Set up Python 3.9.10
+      uses: actions/setup-python@v2
```

Comment:
Not quite sure whether this version should be used as `actions/setup-python@v3` introduces native caching mechanism for poetry (so maybe next step could be avoided)
https://github.com/actions/setup-python/releases/tag/v3.1.0

Reply:
Skipped for a moment, as current updated way of installing poetry works, but we will look into it